### PR TITLE
Change leads_to consistency check to check for 'not disjoint'

### LINKED ISSUE #59299: Obsolete the consistency check for 'leads_to' mutations.
### Is your feature request related to a problem? Please describe.

I made a PR that removes the prereq `EYEBULGE` from the `MEMBRANE`-mutation, but I intended to leave the `"leads_to": [ "MEMBRANE" ],` since I think, it's a good thing, that `leads_to` is a means to modify the weight of the roll towards the next mutation.

However: After removing `EYEBULGE` from lizard mutation it gave me this error:
```
 DEBUG    : mutation EYEBULGE lacks category LIZARD present in additive mutation MEMBRANE

 FUNCTION : check_consistency
 FILE     : D:\a\Cataclysm-DDA\Cataclysm-DDA\src\mutation_data.cpp
 LINE     : 731
 VERSION  : 8157ece
```

### Solution you would like.

I suggest to obsolete the consistency check for `leads_to` as it adds more flexibilty upon designing new mutations and modifying existing ones.

### Describe alternatives you have considered.

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Stadler76**
#### Summary
Bugfixes "Change leads_to consistency check to check for 'not disjoint'"

#### Purpose of change
Closes #59299

The consistency check for `leads_to` required, that all categories in the additive mutation are present in the mutation that `leads_to` it. This was undesirable and therefore it now checks, if the categories of both mutations are not disjoint aka any match is fine. Obsolete mutations aka mutations without a category are skipped.

#### Describe the solution
Made the consistency check for `leads_to`-mutations to check, if at least one category is present in both the mutation and its addition(s).

- [x] Undo the changes to `mutations.json` once the checks give me a thumbs up and convert the draft to a PR.

#### Describe alternatives you've considered
The alternative would be removing `leads_to` upon removing `prereqs` when I just want to remove a category from the 'not prereq anymore'-mutation, which is also undesirable, I'd say.

#### Testing
1. Removed `"prereqs": [ "EYEBULGE" ],` from `MEMBRANE` and `LIZARD`-category from `EYEBULGE` and it succeeded as expected.
2. In addition to the first test: Removed `FISH` and `BATRACHIAN` from `MEMBRANE` and thus making the categories of `MEMBRANE` and `EYEBULGE` disjoint and it failed as expected.

### CODE REVIEW

File: src/mutation_data.cpp [hunk 1]
```diff
@@ -725,11 +725,20 @@ void mutation_branch::check_consistency()
         }
         for( const trait_id &addition : mdata.additions ) {
             const mutation_branch &adata = addition.obj();
+
+            // Skip obsolete additions aka additions without any category.
+            if( adata.category.empty() ) {
+                continue;
```
Comment (actually-a-cat) [inline src/mutation_data.cpp:731]:
```suggestion
            if( mdata.category.empty() ) {
                break;
```

This is equivalent to previous behavior in regard to those obsoleted mutations and also generally a little bit safer (will catch a mistake in which a mutation somehow ends up without category but still has `leads_to` from a normal mutation)

**Review by actually-a-cat — CHANGES_REQUESTED** [review]
Aside from this LGTM. For context, with #58803,

**Comment by Stadler76** [comment]
@actually-a-cat: For some reason I can't Re-request review from you. But can you approve those changes?

**Review by actually-a-cat — APPROVED** [review]

**Comment by Stadler76** [comment]
If anyone wants to know, why I'm merging again after a few hours. The answer is bad timing:
grafik
grafik

**Pull request merged**